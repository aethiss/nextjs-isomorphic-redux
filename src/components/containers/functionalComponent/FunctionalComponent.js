import React, { useState, useEffect } from 'react'

// Lib
import { detectDevice } from '../../../libs/detectDevice'
import styles from '../../../../styles/Home.module.css'

const FunctionalComponent = () => {
  const [mobile, setMobile] = useState(false)
  const [pageLoaded, setPageLoad] = useState(false)

  useEffect(() => {
    if (!mobile) {
      // eslint-disable-next-line no-undef
      const userAgent = window?.navigator.userAgent
      detectDevice(userAgent).then((deviceDetails) => {
        // eslint-disable-next-line no-undef
        setTimeout(() => {
          setMobile(deviceDetails.isPhone)
          setPageLoad(true)
        }, 1000)
      })
    }
  }, [])

  return (
    <main className={styles.main}>
      {pageLoaded ? (
        <div>
          <p>This is Client Side Page</p>
          <p>
            You viewing this page with {mobile ? `mobile` : `desktop`} device !
          </p>
        </div>
      ) : (
        <div>loading ...</div>
      )}
    </main>
  )
}

export default FunctionalComponent
