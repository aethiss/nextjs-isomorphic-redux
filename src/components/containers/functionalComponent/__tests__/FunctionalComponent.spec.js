import React from 'react'
import TestRenderer from 'react-test-renderer'
import FunctionalComponent from '../FunctionalComponent'

jest.useFakeTimers()

import { detectDevice } from '../../../../libs/detectDevice'
jest.mock('../../../../libs/detectDevice', () => {
  // Works and lets you check for constructor calls:
  return {
    detectDevice: jest.fn().mockImplementation(() => {
      // eslint-disable-next-line no-undef
      return new Promise((resolve) => {
        resolve({ isPhone: false, device: undefined })
      })
    }),
  }
})

describe('FunctionalComponent', () => {
  describe('Render', () => {
    it('[FunctionalComponent]: Should render loading default page', () => {
      const mobileRender = TestRenderer.create(<FunctionalComponent />)
      expect(mobileRender).toMatchSnapshot()
    })
    it('[FunctionalComponent]: Should render desktop view', () => {
      TestRenderer.create(<FunctionalComponent />)
      expect(detectDevice).toHaveBeenCalled()
    })
  })
})
