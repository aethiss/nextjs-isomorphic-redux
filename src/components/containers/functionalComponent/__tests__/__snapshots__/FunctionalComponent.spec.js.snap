// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FunctionalComponent Render [FunctionalComponent]: Should render desktop view 1`] = `
<main
  className="main"
>
  <div>
    loading ...
  </div>
</main>
`;

exports[`FunctionalComponent Render [FunctionalComponent]: Should render loading default page 1`] = `
<main
  className="main"
>
  <div>
    loading ...
  </div>
</main>
`;
